app/controllers/api_export_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+# ApiExportController: for exporting subjects via the API
+class ApiExportController < ActionController::API
```

Comment:
I'm not sure I understand, are you proposing creating another `ApplicationController` for API usage?  These API controllers don't need things normally useful for browsers.

Reply:
I am proposing:

```ruby
# app/controllers/application_api_controller.rb
class ApplicationAPIController < ActionController::API; end

# app/controllers/api/foo.rb
class Foo < ApplicationAPIController
 def get
    return Foo.find(params[:id])
  end
end

Reply:
Ok I see, and generally is that just a stylistic thing? That you usually don't inherit directly from `ActionController::API`, but usually define your own `ApplicationController` (or `ApplicationAPIController`) and inherit from that?

Reply:
Yes. It allows for generic methods to be placed in the `ApplicationAPIController` just like we do `ApplicationController`. Something like 

```ruby
  def set_client_app
    @current_client_app = OauthApplication.find_by(id: doorkeeper_token&.application_id)
    head :unauthorized if @current_client_app.nil?
  end
```
May be a candidate that gets placed there since I could see that being used across many different API controllers even if it's only in use by one right now..

Reply:
Gotcha, that makes sense. I will add this, and I think some of our responses like:
```
  def status_not_acceptable
    respond_to do |format|
      format.any { head :not_acceptable }
    end
  end
```
could/should be extracted into there so we are doing them the same way in our two controllers.

Reply:
I'm good with this as well

Reply:
Added this class and factored out some methods in https://github.com/SaraAlert/SaraAlert/pull/823/commits/e541f149fac362f6255473ec723c7cbb53016987.